From 233e9dc6dff603248626827ec2a046df4c97ff02 Mon Sep 17 00:00:00 2001
From: yvanliang <yvan6liang@outlook.com>
Date: Fri, 9 Oct 2026 15:42:14 +0800
Subject: [PATCH 46/60] GSI Android 17 compatibility:
 0048-Add-a-prop-to-disable-animation-in-for-sysui-media-c.patch

Change-Id: Ic50fb3e1d90f824f622da70a955789919e460522
---
 .../media/controls/ui/controller/MediaCarouselController.kt    | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/packages/SystemUI/src/com/android/systemui/media/controls/ui/controller/MediaCarouselController.kt b/packages/SystemUI/src/com/android/systemui/media/controls/ui/controller/MediaCarouselController.kt
index b3b7b027b..1531c966c 100644
--- a/packages/SystemUI/src/com/android/systemui/media/controls/ui/controller/MediaCarouselController.kt
+++ b/packages/SystemUI/src/com/android/systemui/media/controls/ui/controller/MediaCarouselController.kt
@@ -23,6 +23,7 @@ import android.content.Intent
 import android.content.res.ColorStateList
 import android.content.res.Configuration
 import android.database.ContentObserver
+import android.os.SystemProperties
 import android.os.UserHandle
 import android.provider.Settings
 import android.provider.Settings.ACTION_MEDIA_CONTROLS_SETTINGS
@@ -984,7 +985,7 @@ constructor(
                         desiredHostState.showsOnlyActiveMedia
 
                 for (mediaPlayer in MediaPlayerData.players()) {
-                    if (animate) {
+                    if (animate && !SystemProperties.getBoolean("persist.sys.phh.no_media_carousel_animation", false)) {
                         mediaPlayer.mediaViewController.animatePendingStateChange(
                             duration = duration,
                             delay = startDelay,
